# CFSB Coding Challenge
## Question 1a.
### a. Using a brute-force search, find all 4 consecutive 3-digit integers that are like this.

In [35]:
wholeStack=[]  #This has all the numbers which satisfy the condition
currentStack=[]  #This is the current stack which satisfy the condition
Sum=0
for i in range(100,1000):
    
    num=i
    #loop to get the sum of the digits
    while num>0: 
        Remainder=num % 10
        Sum+=Remainder
        num=num//10
    
    #Check if the number is a multiple of the sume of its digits
    if i%Sum==0:
        currentStack.append(i)
        if len(currentStack)==2:
            wholeStack.append(currentStack)
    else:
        currentStack=[]
    Sum=0
print(wholeStack)

[[110, 111, 112], [132, 133], [152, 153], [200, 201], [209, 210], [224, 225], [399, 400], [407, 408], [440, 441], [480, 481], [510, 511, 512, 513], [629, 630], [644, 645], [735, 736], [800, 801], [803, 804], [935, 936]]


### b. Are there 17 consecutive 3-digit integers, such that they cannot be divided by the sum of their digits? 18?
### There are 17.  
###  [[110, 111, 112], [132, 133], [152, 153], [200, 201], [209, 210], [224, 225], [399, 400], [407, 408], [440, 441], [480, 481], [510, 511, 512, 513], [629, 630], [644, 645], [735, 736], [800, 801], [803, 804], [935, 936]]



In [37]:
### c. If your search range was expanded to [10^2 , 10^12 ] – Please discuss how would you speed
### up or optimize your search algorithm without any additional computational resources?
### What utilities, functions or packages would you use during testing to benchmark
### performance?
